build-system/compile/compile.js

```diff
@@ -389,23 +380,29 @@ function compile(
         .on('error', (err) =>
           handleCompilerError(err, outputFilename, options, resolve)
         )
-        .pipe(rename(outputFilename))
+        .pipe(rename(`${outputDir}/${outputFilename}`))
         .pipe(
           gulpIf(
             !argv.pseudo_names && !options.skipUnknownDepsCheck,
             checkForUnknownDeps()
           )
         )
         .on('error', reject)
-        .pipe(sourcemaps.write('.'))
         .pipe(
           gulpIf(
             shouldAppendSourcemappingURLText,
             gap.appendText(`\n//# sourceMappingURL=${outputFilename}.map`)
           )
         )
-        .pipe(postClosureBabel(outputDir))
-        .pipe(gulp.dest(outputDir))
+        .pipe(postClosureBabel())
+        .pipe(
+          sourcemaps.write('.', {
+            sourceRoot: isProdBuild
```

Comment:
I don't think `isProdBuild` works as originally designed any more. It is based on `argv.type`, which could be either `canary` or `prod`. Today, we don't use `--type prod` anywhere, and the release automation uses `--type canary` (only for sourcemaps).

Can we change this logic to key off of `--fortesting` instead? So if `gulp dist` is called normally, sourcemaps will point to `https://raw.githubusercontent.com` and if `--fortesting` is used, they will point to `http://localhost:8000`?

```js
sourceRoot: argv.fortesting
  ? 'http://localhost:8000/',
  : `https://raw.githubusercontent.com/ampproject/amphtml/${internalRuntimeVersion}/`
```